crates/svm-runtime-c-api/src/api.rs

```diff
@@ -835,7 +874,24 @@ pub unsafe extern "C" fn svm_runtime_destroy(runtime: *mut c_void) {
 #[must_use]
 #[no_mangle]
 pub unsafe extern "C" fn svm_imports_destroy(imports: *const c_void) {
-    let _ = Box::from_raw(imports as *mut Vec<Import>);
+    let _ = Box::from_raw(imports as *mut Vec<ExternImport>);
+}
+
+/// Allocates a new `svm_trap_t` with inner `error` of size `size`.
+#[no_mangle]
+pub unsafe extern "C" fn svm_trap_alloc(size: u32) -> *mut svm_trap_t {
```

Comment:
I think that a simpler design would be to accept `svm_byte_array` msg and to re-allocate it on the heap using Rust (similarly to wasmer's `wasm_trap_new`). Copying bytes to an unsafe buffer should work in Go, but it's quite hacky.